imports/plugins/core/orders/client/components/OrderRefunds.js

```diff
@@ -0,0 +1,303 @@
+  const { classes, order } = props;
+  const { payments } = order;
+
+  // useRef
+  const inputLabel = useRef(null);
+
+  // useState
+  const [labelWidth, setLabelWidth] = useState(0);
+  const [refundReasonSelectValues, setRefundReasonSelectValues] = useState({ reason: "" });
+  const [refundTotal, setRefundTotal] = useState(0.00);
+
+  // useEffect
+  // update label width when refund select is activate
+  useEffect(() => {
+    if (inputLabel && inputLabel.current) {
+      setLabelWidth(inputLabel.current.offsetWidth);
+    }
+  }, []);
+
+  const canRefund = payments.every((payment) => payment.status !== "refunded");
+  // calculated refund total from inputs
+  const calculatedRefundTotalDisplay = formatMoney(refundTotal, order.currencyCode);
+  // previous refunds
+  const orderPreviousRefundTotal = order.refunds.reduce((acc, refund) => acc + refund.amount.amount, 0);
+  // available to refund
+  const orderAmountAvailableForRefund = order.summary.total.amount - orderPreviousRefundTotal;
+  const orderAmountAvailableForRefundDisplay = formatMoney(orderAmountAvailableForRefund, order.currencyCode);
+
+  const handleCreateRefund = (data, mutation) => {
+    const { amounts } = data;
+    const { reason } = refundReasonSelectValues;
+
+    // turn form data into an array of payments that provide paymentID and amount
+    // then filter out any amounts that are `null` or `0`
+    const paymentsToRefund = Object.keys(amounts).map((paymentId) => ({
+      paymentId,
+      amount: parseFloat(amounts[paymentId], 10)
+    })).filter((payment) => payment.amount && payment.amount > 0);
+
+    paymentsToRefund.forEach((payment) => {
+      const variables = {
+        amount: payment.amount,
+        orderId: order._id,
+        paymentId: payment.paymentId
+      };
+
+      // Stripe will not accept an empty string or `null` value for the `reason` field,
+      // so only include `reason` in the mutation only if its' value is set
+      if (reason) {
+        variables.reason = reason;
+      }
+
+      if (hasPermission) {
+        mutation({
+          variables
+        });
+      }
+    });
+  };
+
+  // When refund amounts are changed, add up amounts to display in button
+  const handleRefundTotalUpdate = () => {
+    if (this.form && this.form.state) {
+      const { amounts } = this.form.state.value;
+
+      const reducedRefundTotal = Object.keys(amounts).map((paymentId) => ({
+        paymentId,
+        amount: parseFloat(amounts[paymentId], 10)
+      })).filter((payment) => payment.amount && payment.amount > 0).reduce((acc, value) => acc + value.amount, 0);
+
+      setRefundTotal(() => reducedRefundTotal);
+    }
+  };
+
+  const handleRefundAmountChange = (amount) => {
+    if (amount) handleRefundTotalUpdate();
+  };
+
+  const handleRefundReasonSelectChange = (event) => {
+    handleRefundTotalUpdate();
```

Comment:
It doesn't seem like changing the select value would have any impact on `refundTotal` value?

Reply:
Correct. This is leftover from some old code, missed it while taking it out.